src/hooks/useClaimOrderCallback.ts

```diff
@@ -186,10 +195,45 @@ export const useClaimOrderCallback = (
     })
 
     addTransaction(response, {
-      summary: 'Claiming tokens',
+      summary: `Claiming tokens auction-${auctionId}`,
     })
     return response.hash
   }, [account, addTransaction, chainId, error, gasPrice, library, auctionId, claimInfo])
 
-  return claimCallback
+  const claimableOrders = claimInfo?.sellOrdersFormUser
+  const userHasAvailableClaim = useUserHasAvailableClaim(auctionIdentifier, claimableOrders)
+  const pendingApproval = useHasPendingClaim(auctionIdentifier.auctionId, account)
+
+  const claimStatus = useMemo(() => {
+    if (!claimableOrders) return ClaimState.UNKNOWN
+
+    return userHasAvailableClaim
+      ? pendingApproval
+        ? ClaimState.PENDING
+        : ClaimState.NOT_CLAIMED
+      : ClaimState.CLAIMED
+  }, [claimableOrders, pendingApproval, userHasAvailableClaim])
+
+  return [claimStatus, claimCallback]
+}
+
+export function useUserHasAvailableClaim(
+  auctionIdentifer: AuctionIdentifier,
+  claimableOrders?: string[],
+): boolean | undefined {
+  const { account, library } = useActiveWeb3React()
+  const { auctionId, chainId } = auctionIdentifer
+
+  const easyAuctionInstance: Contract = getEasyAuctionContract(chainId as ChainId, library, account)
+  const {
+    loading: isLoadingClaimed,
+    result: not_claimed,
+  } = useSingleCallResult(easyAuctionInstance, 'containsOrder', [
+    auctionId,
+    claimableOrders == undefined || claimableOrders[0] == undefined
```

Comment:
if (claimableOrders == undefined || claimableOrders[0] == undefined) is true, then we don't need to make a call to the blockchain at all. Its sure that this will always return false

Reply:
I see. I am also not sure how to do it better.